### PR TITLE
Merge CE `setImmediate` docs into README

### PR DESCRIPTION
**Pull request by armanbilge**
Tried to keep the best of both explanations, while adjusting for the wider audience here!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -44,16 +44,66 @@ loop()
...
+        setTimeout(() => {
+          // this one (and all after it) are clamped!
+        }, 0);
+      }, 0);
+    }, 0);
+  }, 0);
+}, 0);
+```
+
+Each timeout sets a new timeout, and so on and so on. This is exactly the sort of situation that we get into when chaining `Future`s, where each `map`/`flatMap`/`transform`/etc. schedules a another `Future` which, in turn will schedule another... etc. etc. This is exactly where we see clamping. In particular, the innermost `setTimeout` in this example will be clamped to 4 milliseconds (meaning there is no difference between `setTimeout(.., 0)` and `setTimeout(.., 4)`), which would slow down execution *even more*.
```
[the remark this answers is not in the document]
Comment (armanbilge) [reply README.md:71]:
```suggestion
Each timeout sets a new timeout, and so on and so on. This is exactly the sort of situation that we get into when chaining `Future`s, where each `map`/`flatMap`/`transform`/etc. schedules another `Future` which, in turn will schedule another... etc. etc. This is exactly where we see clamping. In particular, the innermost `setTimeout` in this example will be clamped to 4 milliseconds (meaning there is no difference between `setTimeout(.., 0)` and `setTimeout(.., 4)`), which would slow down execution *even more*.
```

File: README.md [hunk 2]
```diff
@@ -44,16 +44,66 @@ loop()
...
 
-This explains why the above snippet will run forever on a `Promise`-based executor: the microtask queue is *never* empty because we're constantly adding new tasks! Thus, `setTimeout` is never able to run because the macrotask queue never receives control.
+This explains why the above snippet will run forever on a `Promise`-based executor: the microtask queue is *never* empty because we're constantly adding new tasks! Thus, `setTimeout` is never able to run because the macrotask queue never receives control. This is horrible, even by JavaScript standards.
```
Comment (sjrd) [inline README.md:47]: Could we avoid being condescending at JavaScript, please? Scala.js doesn't make that kind of judgment calls. I don't think this new sentence brings any positive value.

File: README.md [hunk 3]
```diff
@@ -44,16 +44,66 @@ loop()
...
+
+You can read more details [in the MDN documentation](https://developer.mozilla.org/en-US/docs/Web/API/WindowTimers.setTimeout#Minimum.2F_maximum_delay_and_timeout_nesting).
+
+### `setImmediate`
+
+Fortunately, we aren't the only ones to have this problem. What we *want* is something which uses the macrotask queue (so we play nicely with `setTimeout`, I/O, and other macrotasks), but which doesn't have as much overhead as `setTimeout`. The answer is `setImmediate`.
+
+The `setImmediate` function was first introduced in NodeJS, and its purpose is to solve *exactly* this problem: a faster `setTimeout(..., 0)`. In particular, `setImmediate(...)` is *semantically* equivalent to `setTimeout(0, ...)`, except without the associated clamping: it doesn't include a delay mechanism of any sort, it simply takes a callback and immediately submits it to the event loop, which in turn will run the callback as soon as its turn comes up.
```
Comment (sjrd) [inline README.md:79]:
```suggestion
The `setImmediate` function was first introduced in Node.js, and its purpose is to solve *exactly* this problem: a faster `setTimeout(..., 0)`. In particular, `setImmediate(...)` is *semantically* equivalent to `setTimeout(0, ...)`, except without the associated clamping: it doesn't include a delay mechanism of any sort, it simply takes a callback and immediately submits it to the event loop, which in turn will run the callback as soon as its turn comes up.
```
There are more of those in the page.

File: README.md [hunk 4]
```diff
@@ -44,16 +44,66 @@ loop()
...
+
+Unfortunately, `setImmediate` isn't available on every platform. For reasons of... their own, Mozilla, Google, and Apple have all strenuously objected to the inclusion of `setImmediate` in the W3C standard set, despite the proposal (which originated at Microsoft) and obvious usefulness. This in turn has resulted in an all-too familiar patchwork of inconsistency across the JavaScript space.
+
+That's the bad news. The good news is that all modern browsers include *some* sort of functionality which can be exploited to emulate `setImmediate` with similar performance characteristics. In particular, *most* environments take advantage of `postMessage` in some way. If you're interested in the nitty-gritty details of how this works, you are referred to [this excellent readme](https://github.com/YuzuJS/setImmediate#the-tricks).
+
+scala-js-macrotask-executor implements *most* of the `setImmediate` polyfill in terms of ScalaJS, wrapped up in an `ExecutionContext` interface. The only elements of the polyfill which are *not* implemented are as follows:
```
Comment (sjrd) [inline README.md:85]:
```suggestion
scala-js-macrotask-executor implements *most* of the `setImmediate` polyfill in terms of Scala.js, wrapped up in an `ExecutionContext` interface. The only elements of the polyfill which are *not* implemented are as follows:
```
There are more of those in the page.

File: README.md (hunk 2 above)
Comment (armanbilge) [reply README.md:47]: I'm sorry, I knew better but I shoe-horned this in anyway. Besides adjusting for the new audience I forgot to adjust for the new venue.

I suppose line 81 below should be dropped/reworked too? Anything else?

File: README.md [hunk 5]
```diff
@@ -13,7 +13,7 @@ libraryDependencies += "org.scala-js" %%% "scala-js-macrotask-executor" % "1.0.0
 Published for Scala 2.11, 2.12, 2.13, 3. Functionality is fully supported on all platforms supported by Scala.js (including web workers). In the event that a given platform does *not* have the necessary functionality to implement `setImmediate`-style yielding (usually `postMessage` is what is required), the implementation will transparently fall back to using `setTimeout`, which will drastically inhibit performance but remain otherwise functional.
 
 ```scala
-import org.scalajs.macrotaskexecutor.MacrotaskExecutor.Implicits._
+import org.Scala.js.macrotaskexecutor.MacrotaskExecutor.Implicits._
```
Comment (sjrd) [inline README.md:16]: Search-and-replace artifact ;)

Comment (armanbilge) [reply README.md:16]: 😳 seems I leaked some of my internal implementation details 😉

Comment (armanbilge) [reply README.md:16]:
```suggestion
import org.scalajs.macrotaskexecutor.MacrotaskExecutor.Implicits._
```

**Review by sjrd — APPROVED** [review]

**Pull request merged**